### ISSUE TITLE
[Proposal] Package reorganization

### ISSUE THREAD

**Issue by pringshia** [facebookresearch/Mephisto#501]
| Status:   |  Draft (please feel free to edit/comment/rewrite/etc)  |
| ------------- | ------------- |

## Overview

Sets up a strategy for publishing smaller packages for the annotation-toolkit. 

Currently it's:

```jsx
npm install annotation-toolkit
npm install bootstrap-chat

//^ a catch-all package that encompasses all portions of the library
```

We propose something like:
```jsx
npm install @annotations/layout
npm install @annotations/video-player
npm install @annotations/bounding-box
npm install @annotations/keypoints
npm install @annotations/chat
```

here `@annotation` is just used for illustrative purposes as an npm package name. the actual name would be something else.

## Top of Mind

Backwards compatibility will be a key priority for us and we'll continue to support the old packages for existing users. We'll also map out a migration plan to help existing users upgrade as needed.

We think this package reorganization is better to do early rather than later so that we can our incorporate early learnings in quickly while the change cost is still low.

The benefits here include a clearer path to integration and shareability for component collaborators.
This will help us build up our repository of [shared components](https://annotation-toolkit-storybook.vercel.app/?path=/story/example-0-introduction--page).

## Effort

We need to figure out:
- the right package divisions
- how to avoid duplication across packages (can we extract out build steps, code linting, etc? what is the effort in doing this?)
- what is the migration strategy for existing users?
- how to correctly set up a js org

**Comment by JackUrb** [comment]
Definitely excited about this! Interested to know how we'd manage packages for downstream teams, but I'm sure that's something to be sketched out as we understand our users.

**Comment by pringshia** [comment]
Potential deploy strategy inspiration: https://github.com/radix-ui/primitives

**Comment by JackUrb** [comment]
Wow `radix` looks awesome, both in their deploy style (which I think is right in line with what you were considering) and also in having some nice components we may use here and there. Good find

**Comment by pringshia** [comment]
Some more inspo for a design system monorepo with multiple packages. Carbon by IBM: https://github.com/carbon-design-system/carbon

**Comment by JackUrb** [comment]
This is done now, right?

**Comment by pringshia** [comment]
Yep, merged with #565. Closing.
